src/lib.rs

```diff
@@ -62,6 +63,28 @@ pub enum RunError<E> {
     TimedOut,
 }
 
+impl<E> fmt::Display for RunError<E>
+    where E: error::Error + 'static
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            RunError::User(ref err) => write!(f, "User error: {}", err),
+            RunError::TimedOut => write!(f, "Timed Out"),
```

Comment:
I think it would be better to pass through the user error as is and then change timeout to something like "Timed out in bb8"